File: app/schemas/copilot.py
"""面向 Vue 求职副驾的会话与分析产物契约。"""

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


CopilotRole = Literal["user", "assistant"]
TurnStatus = Literal["pending", "running", "completed", "failed"]
ArtifactType = Literal["job_brief", "evidence_map", "fit_strategy", "action_bundle"]
ArtifactDecision = Literal["accept", "reject", "ask", "create_task"]


class CopilotSessionCreate(BaseModel):
    resume_version_id: int | None = Field(default=None, gt=0)
    target_role: str = Field(default="", max_length=120)


class CopilotSessionResponse(BaseModel):
    id: int
    resume_version_id: int | None = None
    active_report_id: int | None = None
    target_role: str = ""
    status: str
    created_at: datetime
    updated_at: datetime


class CopilotMessageCreate(BaseModel):
    content: str = Field(min_length=2, max_length=20000)


class CopilotMessageResponse(BaseModel):
    id: int
    session_id: int
    role: CopilotRole
    content: str
    turn_id: int | None = None
    created_at: datetime


class AnalysisArtifactResponse(BaseModel):
    id: int
    turn_id: int
    artifact_type: ArtifactType
    payload: dict[str, Any]
    status: str
    created_at: datetime


class AnalysisTurnResponse(BaseModel):
    id: int
    session_id: int
    status: TurnStatus
    stage: str = "queued"
    progress: int = Field(default=0, ge=0, le=100)
    error_message: str = ""
    report_id: int | None = None
    parent_turn_id: int | None = None
    input_type: Literal["initial_jd", "follow_up"] = "initial_jd"
    created_at: datetime
    updated_at: datetime
    artifacts: list[AnalysisArtifactResponse] = Field(default_factory=list)


class CopilotSessionDetailResponse(CopilotSessionResponse):
    messages: list[CopilotMessageResponse] = Field(default_factory=list)
    turns: list[AnalysisTurnResponse] = Field(default_factory=list)


class ArtifactDecisionCreate(BaseModel):
    decision: ArtifactDecision
    note: str = Field(default="", max_length=2000)


class ArtifactDecisionResponse(BaseModel):
    id: int
    artifact_id: int
    decision: ArtifactDecision
    note: str = ""
    created_at: datetime

File: frontend-web/src/api/copilot.ts
import { apiFetch } from "./client";

export type ArtifactType = "job_brief" | "evidence_map" | "fit_strategy" | "action_bundle";
export type TurnStatus = "pending" | "running" | "completed" | "failed";
export type TurnInputType = "initial_jd" | "follow_up";

export interface ResumeVersion {
  id: number;
  version_name: string;
  target_role: string;
  raw_text: string;
  profile: {
    education: string[];
    skills: string[];
    projects: Array<{
      name: string;
      role: string;
      technologies: string[];
      description: string;
      achievements: string[];
    }>;
    internships: string[];
    awards: string[];
    target_roles: string[];
    available_from: string;
    parse_notes: string[];
  };
  created_at: string | null;
}

export interface CopilotMessage {
  id: number;
  session_id: number;
  role: "user" | "assistant";
  content: string;
  turn_id: number | null;
  created_at: string;
}

export interface Artifact {
  id: number;
  turn_id: number;
  artifact_type: ArtifactType;
  payload: Record<string, unknown>;
  status: string;
  created_at: string;
}

export interface AnalysisTurn {
  id: number;
  session_id: number;
  status: TurnStatus;
  stage: string;
  progress: number;
  error_message: string;
  report_id: number | null;
  parent_turn_id: number | null;
  input_type: TurnInputType;
  created_at: string;
  updated_at: string;
  artifacts: Artifact[];
}

export interface CopilotSession {
  id: number;
  resume_version_id: number | null;
  active_report_id: number | null;
  target_role: string;
  status: string;
  created_at: string;
  updated_at: string;
  messages: CopilotMessage[];
  turns: AnalysisTurn[];
}

export function listResumeVersions(): Promise<ResumeVersion[]> {
  return apiFetch("/api/resumes/versions");
}

export function createSession(payload: { resume_version_id?: number; target_role?: string }) {
  return apiFetch<CopilotSession>("/api/v1/copilot/sessions", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function getSession(sessionId: number) {
  return apiFetch<CopilotSession>(`/api/v1/copilot/sessions/${sessionId}`);
}

export function sendMessage(sessionId: number, content: string) {
  return apiFetch<AnalysisTurn>(`/api/v1/copilot/sessions/${sessionId}/messages`, {
    method: "POST",
    body: JSON.stringify({ content }),
  });
}

export function getTurn(turnId: number) {
  return apiFetch<AnalysisTurn>(`/api/v1/copilot/turns/${turnId}`);
}

export function decideArtifact(
  artifactId: number,
  decision: "accept" | "reject" | "ask" | "create_task",
) {
  return apiFetch(`/api/v1/copilot/artifacts/${artifactId}/decisions`, {
    method: "POST",
    body: JSON.stringify({ decision }),
  });
}
